torch/_prims/nvfuser_executor.py

```diff
@@ -119,3 +121,61 @@ def nvfuser_execute(gm: GraphModule, *args):
         fusion.execute(concrete_fusion_inputs),  # type: ignore[has-type]
         unflatten_spec,  # type: ignore[has-type]
     )
+
+
+class NvfuserPrimOperatorSupport(torch.fx.passes.operator_support.OperatorSupport):
+    def is_node_supported(self, submodules, node: torch.fx.Node) -> bool:
+        return (
+            node.op == "call_function"
+            and getattr(node.target, "impl_nvfuser", None) is not None
+        )
+
+
+class PartitionedInterpreter(torch.fx.Interpreter):
+    def call_module(self, target, args, kwargs):
+        assert isinstance(target, str)
+        assert len(kwargs) == 0
+        submod = self.fetch_attr(target)
+        # CapabilityBasedPartitioner hardcodes the name of the subgraphs with supported_ops as "fused_" + subgraph id
+        if target.startswith("fused_"):
+            return nvfuser_execute(submod, *args)
+        else:
+            return super().call_module(target, args, kwargs)
+
+
+# MyPy bug: https://github.com/python/mypy/issues/5107
+@lru_cache()  # type: ignore[arg-type]
+def maybe_partition_graph(gm: GraphModule):
+    supported_ops = NvfuserPrimOperatorSupport()
+    call_function_nodes = filter(lambda n: n.op == "call_function", gm.graph.nodes)
+    # the graph is partitioned only if at least one node is not supported by nvFuser
+    any_unsupported = any(
+        not supported_ops.is_node_supported(None, node) for node in call_function_nodes
+    )
+    if any_unsupported:
+        # CapabilityBasedPartitioner modifies the graph in-place so we need to make a copy of the graph
+        gm = deepcopy(gm)
+        partitioner = CapabilityBasedPartitioner(
+            gm, supported_ops, allows_single_node_partition=True
+        )
+        partitions = partitioner.propose_partitions()
+        if len(partitions) == 0:
+            raise RuntimeError(
```

Comment:
nitpick, error out feels like a confusing UX. Maybe change this to a warning instead?